src/profile/helpers/editData.ts

```diff
@@ -80,17 +87,42 @@ export type EditData = {
   readonly saving: SaveType;
 };
 
+type Backups = {
+  add: (item: EditData) => void;
+  clean: (allItems: EditData[]) => void;
+  get: (id: string) => EditData | undefined;
+};
+
 export type EditFunctions = {
   create: (newProfileData: EditDataProfileSource) => EditData;
   getEditData: () => EditData[];
+  updateItemAndCreateSaveData: (
+    targetItem: EditData,
+    newValue: EditDataValue
+  ) => Partial<FormValues>;
+  updateData: (newProfileRoot: ProfileRoot) => boolean;
+  updateAfterSavingError: (id: string) => boolean;
+  resetItem: (targetItem: EditData) => boolean;
 };
 
-function isMultiItemDataType(dataType: EditDataType): boolean {
+export function isMultiItemDataType(dataType: EditDataType): boolean {
   return !(
     dataType === basicDataType || dataType === additionalInformationType
   );
 }
 
+export function isSaving(allItems: EditData[]): boolean {
+  return !!allItems.find(item => !!item.saving);
+}
+
+export function getNewItem(allItems: EditData[]): EditData | undefined {
+  return allItems.find(item => item.id === '');
+}
+
+export function hasNewItem(allItems: EditData[]): boolean {
+  return !!getNewItem(allItems);
+}
```

Comment:
Did not find any usages of these functions outside of this file, so the exports are probably not needed?

Reply:
Correct, they (like other unused exports) were exported for testing, but they are tested via functions that use them.

Reply:
Removed